src/main/java/org/jenkinsci/plugins/tuleap_git_branch_source/client/impl/DefaultClient.java

```diff
@@ -46,17 +47,7 @@
         OkHttpClient.Builder builder = new OkHttpClient.Builder()
             .connectTimeout(10, TimeUnit.SECONDS).writeTimeout(10, TimeUnit.SECONDS).readTimeout(10, TimeUnit
                 .SECONDS).cache(null);
-        if (this.credentials.isPresent()) {
-            final StandardCredentials c = this.credentials.get();
-            if (c instanceof StandardUsernamePasswordCredentials) {
-                final String username = ((StandardUsernamePasswordCredentials) c).getUsername();
-                final String password = ((StandardUsernamePasswordCredentials) c).getPassword().getPlainText();
-                builder.addInterceptor(new BasicAuthInterceptor(username, password));
-            } else {
-                throw new UnsupportedOperationException(
-                    "Not implemented yet, only StandardUsernamePasswordCredentials " + "is supported ... for the moment");
-            }
-        }
+        credentials.ifPresent( token -> builder.addInterceptor(new AccessKeyInterceptor(token)));
```

Comment:
[Question] Why do you not check if the credentials type is the good one ?

Reply:
It was necessary before because the function's parameter was an `Optional<StandardCredentals>` and we had to make sure it was `StandardUsernamePasswordCredentials` before dealing with it. Now everything is a `TuleapAccessToken` so the check isn't needed anymore.